### ISSUE TITLE
after switching to 9.0.0.0-preview410690 project won't build

### ISSUE THREAD

**Issue by sjorsmiltenburg** [dotnet/maui#22967]
### Description

New project upgraded to 9.0.0.0-preview410690 project won't compile

maybe related to https://github.com/dotnet/maui/issues/5785

### Steps to Reproduce

- I create a new maui application, and compile and run it for windows it, it runs fine.
- I remove bin and debug dirs
- I then go to dependancies, enable prerelease and update the maui packages to 9.0.0.0-preview410690 
- I try to compile and run again

It can't find MauiWinUIApplication anymore
![image](https://github.com/dotnet/maui/assets/649207/19707722-1a8d-4483-8529-542dc7403719)

The suggestion in the [linked item](https://github.com/dotnet/maui/issues/5785) does not seem to work / apply to my case
![image](https://github.com/dotnet/maui/assets/649207/ce5d18b2-9f50-4ed9-99cd-af7dbe5439be)

![image](https://github.com/dotnet/maui/assets/649207/0208a9e8-a012-461b-a702-c94d52c3c5ab)

I'm running on VS Version 17.10.0 Preview 4.0

### Version with bug

9.0.0-preview.4.10690

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.0-rc.1.9171

### Affected platforms

Windows

### Did you find any workaround?

no

**Comment by sjorsmiltenburg** [comment]
after running 'dotnet workload update' it still does not work
![image](https://github.com/dotnet/maui/assets/649207/20b825de-31a2-40f6-b93e-69214aae0370)
![image](https://github.com/dotnet/maui/assets/649207/2d52e59e-c6c3-42b1-b2ed-171d062847b8)

**Comment by sjorsmiltenburg** [comment]
The build problem still not went away after I
- installed .NET 9.0 SDK (v9.0.100-preview.4) 
- ran 'dotnet workload restore' in the package manager console
- set the Target .Net Runtime in the project to 9.0
 
![image](https://github.com/dotnet/maui/assets/649207/93e427be-87fd-4600-ba14-c5ce22a7b96b)

After these installs a new project now builds and runs.
(a new problem arises though, as when I try to open the project properties, It says it can't load it beacue of missing keys)

Here end my efforts of trying to respond to your "trylatest" call on some of the bugs I've reported.
I'll stick to xamarin.forms for some more months before I try to venture back into maui land. brrr..

**Comment by drasticactions** [comment]
<img width="266" alt="スクリーンショット 2024-06-11 20 45 39" src="https://github.com/dotnet/maui/assets/898335/d27a6104-0ee7-48b6-a95d-1ac37f608efb">

I don't think you updated the target frameworks, looking at what target framework is listed in that XAML. You can't mix and match net9 packages with net8 target frameworks. If you bump those to net9 it should work.

**Comment by sjorsmiltenburg** [comment]
you are correct, maccatalyst (the version that I was not trying to launch) still had it's target framework set to 8.0.
After correcting this it did build.
![image](https://github.com/dotnet/maui/assets/649207/a7f8db4c-422d-4229-a3e2-00d0068694c9)
I found that the 'new problem' was fixed by updating my visual studio version because i was experiencing [this bug](https://github.com/dotnet/maui/issues/21758)

So... all is good now.
Except the whish for better error messages that actually help you get to the cause of the problem.
